File: app/src/main/java/bj/drunkmessengerblocker/main/MainModule.java
package bj.drunkmessengerblocker.main;

import dagger.Module;
import dagger.Provides;

/**
 * Created by j on 15/01/2017.
 */
@Module
public class MainModule
{
    private MainContract.View mainView;

    public MainModule(MainContract.View mainView)
    {
        this.mainView = mainView;
    }

    @Provides
    MainContract.View providesMainContractView()
    {
        return mainView;
    }
}

File: app/src/main/java/bj/drunkmessengerblocker/main/MainModule.kt
package bj.drunkmessengerblocker.main

import dagger.Module
import dagger.Provides

/**
 * Created by Josh Laird on 23/05/2017.
 */
@Module
class MainModule(val view: MainContract.View) {
    @Provides
    fun provideMainContractView(): MainContract.View {
        return view
    }
}